### ISSUE TITLE
Bug in setting member status with any membership transition that is not keyholder

### ISSUE THREAD

**Issue by laftho** [vhs/nomos#175]
laftho [9:25 AM] 
deployed latest

jarrett [9:27 AM] 
cool, so I was just handling a returning member's stuff

[9:27] 
if you take a look at the IPN records, you can see his transactions posted properly

[9:27] 
his account was set to "inactive" though

[9:27] 
I had to set it to active manually

laftho [9:29 AM] 
which IPN?

new messages
jarrett [9:29 AM] 
id 825 to 827

laftho [9:30 AM] 
interesting

[9:31] 
yea I wanted to fix the email crap and my next item was to start handling all these non-payment type transactions

[9:31] 
seeing some new ones there for the first time (for me) such as subr_modify subcr_signup

jarrett [9:34 AM] 
it does look like a new subscription generates three IPNs

laftho [9:35 AM] 
ok yea I see what happened

[9:35] 
this is a bug for members

[9:35] 
https://github.com/vhs/nomos/blob/master/app/processors/PaymentProcessor.php#L164

GitHub
vhs/nomos
nomos - Membership management system made VHS-centric

[9:36] 
https://github.com/vhs/nomos/blob/master/app/services/UserService.php#L251

GitHub
vhs/nomos
nomos - Membership management system made VHS-centric

[9:37] 
I'm assuming this is the scenario:

[9:37] 
- account was a keyholder

[9:37] 
- account was inactive

[9:37] 
- new payment chose member as membership type

[9:38] 
- member != keyholder := it goes to the update membership service

[9:38] 
- membership isn't a keyholder so it doesn't adjust the status to pending

[9:38] 
- back to the processor, never changes the state back to active

[9:39] 
this part of the logic in the processor is really not well laid out

[9:39] 
too much non-obvious stuff happening

[9:40] 
perhaps need a check after that to detect these types of member switching that needs admin attention and send you some emails

jarrett [9:42 AM] 
what you say makes sense, he's not vetted, but maybe he was a friend of VHS, which would still trigger that bug

laftho [9:42 AM] 
ah ok yea if he was a friend that would trigger it yea

[9:42] 
basically we only handle transistions to keyholder

[9:43] 
it'll be good to touch this code to fix this stuff and add tests

[9:43] 
also add the different subscription type handing, etc

**Comment by TyIsI** [comment]
Fixed in #245 #243 and #244
